src/services/completions.ts

```diff
@@ -4299,6 +4298,34 @@ namespace ts.Completions {
             }
             return undefined;
         }
+
+    }
+
+    function getSingleLineReplacementSpanForImportCompletionNode(node: ImportDeclaration | ImportEqualsDeclaration | ImportSpecifier | Token<SyntaxKind.ImportKeyword> | undefined) {
+        if (!node) return undefined;
+        const top = (findAncestor(node, or(isImportDeclaration, isImportEqualsDeclaration)) || node) as ImportDeclaration | ImportEqualsDeclaration | Token<SyntaxKind.ImportKeyword>;
+        const sourceFile = top.getSourceFile();
+        if (rangeIsOnSingleLine(top, sourceFile)) {
+            return createTextSpanFromNode(top, sourceFile);
+        }
+        // Import keyword alone was necessarily on one line
+        Debug.assert(top.kind !== SyntaxKind.ImportKeyword);
+        const withoutModuleSpecifier: TextRange = {
+            pos: top.getFirstToken()!.getStart(),

```

Comment:
```suggestion
            pos: top.getFirstToken(sourceFile)!.getStart(sourceFile),

```

Reply:
It’s necessarily a parse tree node, so this is never needed IIUC.

Reply:
It's more of an efficiency thing - if you don't pass in the `sourceFile` and the children aren't cached, `getFirstToken` needs to run up the parents to get the `sourceFile` to get the source text. Then `getStart` needs to run back up the tree to get the source text as well.

I guess for an import node, it doesn't matter much though. 😅